### PR TITLE
Add more function comments to the rangemapper implementation

### PR DESCRIPTION
**Pull request by chaudum**
**What this PR does / why we need it**:

Helps developers understand how the rangemapper works.

**Checklist**
- [ ] Documentation added
- [ ] Tests updated
- [ ] Add an entry in the `CHANGELOG.md` about the changes.

### CODE REVIEW

File: pkg/logql/rangemapper.go [hunk 1]
```diff
@@ -30,10 +30,16 @@ var splittableRangeVectorOp = map[string]struct{}{
 	syntax.OpRangeTypeMin:       {},
 }
 
+// RangeMapper is used to rewrite LogQL sample expressions into multiple
+// downstream sample expressions with a smaller time range that can be executed
+// using the downstream engine.
+// A rewrite is performed using the following rules:
```
Comment (kavirajk) [inline pkg/logql/rangemapper.go:36]: > A rewrite is performed using the following rules:

but rules are missing?

Comment (kavirajk) [reply pkg/logql/rangemapper.go:36]: nit: A simple example would also be very helpful IMO :) (also for `Parse` and `Map`)

**Review by kavirajk — COMMENTED** [review]
LGTM 👍 . Need to adding missing rules I think.

File: pkg/logql/rangemapper.go (hunk 1 above)
Comment (chaudum) [reply pkg/logql/rangemapper.go:36]: 🤦‍♂️ I wanted to update the comment after writing the comments for `Map()` and `Parse()` and forgot about it.

**Review by ssncferreira — APPROVED** [review]
LGTM ✍️ 
Thank you @chaudum 🙏 it looks really good ❤️ 
Just a couple of minor suggestions

**Pull request merged**